### DISCUSSION TITLE
Nextjs _app always redefine its states with the initial values

### DISCUSSION THREAD

**Discussion by DungRaymond** [vercel/next.js#22490] [Help]
**What version of Next.js are you using?**

10.0.6

**What version of Node.js are you using?**

12

**What browser are you using?**

chrome

**What operating system are you using?**

windows

**How are you deploying your application?**

vercel

**Describe the Bug**

I have a scroll event that will change some states if conditions are satisfied. But it keeps resetting the state value to default which caused performance decline for re-rendering too much and not work as I expected it should be when it's written in pure React.

**Expected Behavior**

The CSR feature on _app.js seems not to work properly

**To Reproduce**

const handleScroll = () => {
if(true) {
  setState(new_value)
  console.log(state)
  }
}
useEffect(() => {
  window.addEventListener('scroll', handleScroll, false)
})

// result: old_value, new_value, old_value, new_value....

**Comment by madeinfree** [comment]
Hi @DungRaymond

From you'r code, I think it is not the bug, you can try to add `[]` for useEffect like

```js
useEffect(() => {
  window.addEventListener('scroll', handleScroll, false)
}, []) // <-- here
```

＊Don't forget to return `removeEventListener`

you can read [here](https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect) to learn more.
